internal/logger/logger.go

```diff
@@ -169,7 +181,7 @@ func (f *loggerFactory) newLogger(level config.LogSeverity) *slog.Logger {
 }
 
 func (f *loggerFactory) createJsonOrTextHandler(writer io.Writer, levelVar *slog.LevelVar, prefix string) slog.Handler {
-	if f.format == "text" {
+	if f.format == textFormat {
```

Comment:
Could f.format ever be anything other than "text" or "json" ? If yes, this logic seems to assume that the default logger format is "json", which should not be assumed, but rather be dependent on a config constant.

Reply:
I'd return an error the user passed format anything other than "text" or "json". Is that checked somewhere, or do we just default to json in that case?

Reply:
Is that checked somewhere, or do we just default to json in that case?

> we default to json in that case. As per our documentation:
```
--log-format VALUE  Specifies the format of the log file. The value can be either text or json. The default value is json.
```

Reply:
I am guessing default here means that if log-format is not set, we assume json. My question was what if user passed a third value. Do we fail in that case ? Or do we pass that same value down here?

Reply:
@ashmeenkaur ^

Reply:
> I am guessing default here means that if log-format is not set, we assume json. My question was what if user passed a third value. Do we fail in that case ? Or do we pass that same value down here?

If a third value is passed, which GCSFuse does not recognizes, the same value is passed here and the logs are written in JSON format. I will raise a separate PR to override it to JSON.